### PR TITLE
feat: support multiple wallet connections

### PR DESCRIPTION
**Pull request by dawidsowardx**

### CODE REVIEW

File: src/chrome/background/background.ts [hunk 1]
```diff
@@ -36,13 +37,17 @@ const handleOnInstallExtension = async () => {
...
+    [key: string]: chrome.storage.StorageChange
+  },
+  area: string,
+) => {
+  if (changes['connections'] && area === 'local') {
+    handleConnectionsChange(changes['connections']?.newValue)
+  }
 
-  if (changes['options']) {
+  if (changes['options'] && area === 'sync') {
```
Comment (xstelea) [inline src/chrome/background/background.ts:50]: Are we syncing options cross devices?

File: src/chrome/messages/message-client.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export const MessageClient = (
     logger: AppLogger
   },
 ) => {
+  const logger = input.logger.getSubLogger({ name: `[MC]:[${origin}]` })
```
Comment (xstelea) [inline src/chrome/messages/message-client.ts:35]: not used

File: src/chrome/offscreen/wallet-connection/factory.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { AppLogger } from 'utils/logger'
+import { WalletConnectionClient } from './wallet-connection-client'
+import { config, radixConnectConfig } from 'config'
+import { MessagesRouter } from 'chrome/offscreen/wallet-connection/messages-router'
+import { ConnectorClient } from '@radixdlt/radix-connect-webrtc'
+import { syncClient } from './sync-client'
+
+export type walletConnectionClientFactory = typeof walletConnectionClientFactory
+
+export const walletConnectionClientFactory = (input: {
```
Comment (xstelea) [inline src/chrome/offscreen/wallet-connection/factory.ts:10]: follow the same name convention as the other clients. `WalletConnectionClientFactory`

File: src/chrome/offscreen/wallet-connection/wallet-connection-client.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+          shouldRetry: false,
+        })),
+    ),
+  })
+
+  const subscription = new Subscription()
+
+  subscription.add(
+    syncClient.dappRequestConfirmedByWallet$.subscribe((interactionId) => {
+      dAppRequestQueue.cancel(interactionId)
```
Comment (xstelea) [inline src/chrome/offscreen/wallet-connection/wallet-connection-client.ts:101]: Should only cancel jobs that are pending.

File: src/chrome/offscreen/wallet-connection/wallet-connection-client.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+      messagesRouter.removeByInteractionId(interactionId)
```
Comment (xstelea) [inline src/chrome/offscreen/wallet-connection/wallet-connection-client.ts:102]: If I understand correctly `messagesRouter.removeByInteractionId(interactionId)` is currently called on all connections. How does the wallet interaction find it's way to the request origin?

File: src/chrome/background/background.ts (hunk 1 above)
Comment (dawidsowardx) [reply src/chrome/background/background.ts:50]: Yes and we always were syncing them. Do you think we shouldn't?

File: src/chrome/offscreen/wallet-connection/wallet-connection-client.ts (hunk 4 above)
Comment (dawidsowardx) [reply src/chrome/offscreen/wallet-connection/wallet-connection-client.ts:101]: Looks like the logic to only cancel pending jobs is already inside the queue

File: src/chrome/offscreen/wallet-connection/wallet-connection-client.ts (hunk 5 above)
Comment (dawidsowardx) [reply src/chrome/offscreen/wallet-connection/wallet-connection-client.ts:107]: You're right, I added it after I tested it 😓 Fixed, now we remove it from `messageRouter` once response is back from mobile.

File: src/chrome/offscreen/message-handler.ts [hunk 6]
```diff
@@ -1,110 +1,75 @@
...
-    sendMessageWithConfirmation: SendMessageWithConfirmation,
-    tabId?: number,
-  ): MessageHandlerOutput => {
+  return (message: Message): MessageHandlerOutput => {
     switch (message?.discriminator) {
-      case messageDiscriminator.walletMessage: {
-        if (isLedgerRequest(message.data)) {
-          logger.debug('🪪 -> 📒: walletToLedgerSubject', message.data)
+      case messageDiscriminator.setConnections: {
+        logger?.debug('setConnections', { message })
```
Comment (xstelea) [inline src/chrome/offscreen/message-handler.ts:30]: Users are posting connector extension logs in public channels. We should not log the connection password

File: src/chrome/offscreen/wallet-connection/wallet-connection-client.ts (hunk 4 above)
Comment (xstelea) [reply src/chrome/offscreen/wallet-connection/wallet-connection-client.ts:101]: Ok great!

File: src/chrome/background/background.ts (hunk 1 above)
Comment (xstelea) [reply src/chrome/background/background.ts:50]: I'm not against it but I recall making a decision to keep it local. Reason is that a user can have different wallet setups on different devices. Maybe just relevant for connections.

Comment (dawidsowardx) [reply src/chrome/background/background.ts:50]: On the other hand you may want to sync your notifications/signaling server settings

File: src/chrome/offscreen/message-handler.ts (hunk 6 above)
Comment (dawidsowardx) [reply src/chrome/offscreen/message-handler.ts]: something in between is smart enough to do this 
Screenshot 2024-01-03 at 15 26 15

**Review by xstelea — APPROVED** [review]

**Pull request merged**